docs/cryptodoc/src/05_10_cmce.rst

```diff
@@ -0,0 +1,469 @@
+required for generating a random sequence :math:`\alpha_0,...,\alpha_{n-1}` of distinct
+:math:`\mathbb{F}_{q}` elements, which is necessary for creating the Goppa
+code. Botan's ``Classic_McEliece_Field_Ordering`` serves as a container for the
+field ordering and includes the algorithm to generate it. To sort the pairs
+:math:`(a_i, i)` (as described in Step 3 of Section 8.2 in [CMCE-ISO]_), a
+constant-time bitonic sort implementation is utilized. This sorting algorithm is
+particularly suitable for sets with a power of two elements.
+
+Another vital role of the field ordering class is managing the Beneš network.
+As Section 9.2.10 of [CMCE-ISO]_ outlines, the Beneš network stores the field
+ordering as control bits in a compact form. Botan employs
+a constant-time implementation of the ``controlbits`` algorithm presented in
+Fig. 7.1 of [CBits]_ to create control bits, and the ``permutation`` algorithm
+to reconstruct the field ordering from the control bits. The creation
+algorithm also utilizes constant-time bitonic sorting.
+
+
+.. _pubkey/cmce/poly:
+
+Polynomial Arithmetic
+^^^^^^^^^^^^^^^^^^^^^
+
+The Classic McEliece algorithm uses elements in the polynomial ring
+:math:`\mathbb{F}_q [y]` in multiple places. The monic irreducible
+polynomial defines the Goppa code, and performing decapsulation requires an
+error locator polynomial. Botan's ``Classic_McEliece_Poly`` class is designed
+to represent these polynomials. It consists of a vector of GF elements, which
+serve as the polynomial coefficients. Additionally, this class provides
+the necessary logic to evaluate the polynomial at a given point in
+:math:`\mathbb{F}_q`.
+
+Polynomial arithmetic is required to obtain the monic irreducible polynomial
+:math:`g` from a random element :math:`\beta \in \mathbb{F}_{q} [y]/F(y)`, as
+described in Step 3 of Section 8.1 in [CMCE-ISO]_. This arithmetic
+is implemented in the ``Classic_McEliece_Polynomial_Ring`` class, representing
+the polynomial ring :math:`\mathbb{F}_{q} [y]/F(y)` and supporting the
+multiplication of two polynomials.
+
+Following the recommendation in the implementors' guide Section 6.1 that
+accompanies [CMCE-R4]_, the minimal polynomial is computed by finding a unique
+solution to the equation :math:`g_0\beta^0 + ... + g_{t-1}\beta^{t-1} = \beta^t`.
+A constant-time Gauss elimination algorithm is used to solve this equation.
+The algorithm aborts if the solution is non-unique. The minimal polynomial
+is then represented as a ``Classic_McEliece_Minimal_Polynomial`` object, which
+includes logic for serialization and deserialization as described in
+Section 9.2.9 of [CMCE-ISO]_.
+
+
+.. _pubkey/cmce/bitvector:
+
+Bit Vector Arithmetic
+^^^^^^^^^^^^^^^^^^^^^
+
+Botan's ``bitvector`` class is a versatile tool for working with bits.
+One such application is the Classic McEliece algorithm, where it handles
+binary matrices, error vectors, code words, and the column selections in the private
```

Comment:
Do you mean the control bits?

Reply:
I think column selection is the correct term (i.e. the result of `move_columns`). We don't really use them for control bit generations.

Reply:
True. I think I meant that we also use `bitvector `for the control bits storage, i.e., we read them as bitvectors and store them as bitvectors. It's minor but maybe we can add "control bit storage" to the sentence.

Reply:
Fair point, I'll add it 👍